packages/frontend-2/components/common/EditableTitleDescription.vue

```diff
@@ -0,0 +1,194 @@
+<template>
+  <div class="my-8">
+    <!-- Editable Title -->
+    <div class="flex group">
+      <label class="max-w-full overflow-hidden">
+        <div class="sr-only">Edit title</div>
+        <div
+          :class="titleInputClasses"
+          class="grow-textarea"
+          :data-replicated-value="title"
+        >
+          <textarea
+            v-model="title"
+            maxlength="512"
+            :class="titleInputClasses"
+            placeholder="Please enter a valid title"
+            rows="1"
+            spellcheck="false"
+            :disabled="isDisabled"
+            :cols="title && title.length < 20 ? title.length : undefined"
+            @keydown="onInputKeydown"
+            @blur="onBlur('title')"
+            @input="onTitleChange"
+          />
+        </div>
+      </label>
+      <PencilIcon
+        v-if="canEdit"
+        class="shrink-0 ml-2 mt-3 w-4 h-4 opacity-0 group-hover:opacity-100 transition text-foreground-2"
+      />
+    </div>
+
+    <!-- Editable Description -->
+    <div class="mt-3 flex gap-x-2 group">
+      <div class="hidden md:inline-block shrink-0 text-foreground-2 mt-0.5">
+        <InformationCircleIcon class="w-5 h-5" />
+      </div>
+      <label>
+        <div class="sr-only">Edit description</div>
+        <div
+          class="grow-textarea"
+          :data-replicated-value="description"
+          :class="descriptionInputClasses"
+        >
+          <textarea
+            v-model="description"
+            :class="[
+              ...descriptionInputClasses,
+              description ? 'focus:min-w-0' : 'min-w-[260px]'
+            ]"
+            :placeholder="description ? undefined : 'Click here to add a description.'"
+            :disabled="isDisabled"
+            rows="1"
+            spellcheck="false"
+            maxlength="1000"
+            :cols="
+              description && description?.length < 20 ? description.length : undefined
+            "
+            @keydown="onInputKeydown"
+            @blur="onBlur('description')"
+            @input="onDescriptionChange"
+          />
+        </div>
+      </label>
+      <div class="shrink-0 ml-2 mt-1 text-foreground-2">
+        <PencilIcon
+          v-if="canEdit"
+          class="w-4 h-4 opacity-0 group-hover:opacity-100 transition text-foreground-2"
+        />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { PencilIcon } from '@heroicons/vue/20/solid'
+import { InformationCircleIcon } from '@heroicons/vue/24/outline'
+import { debounce } from 'lodash-es'
+
+const props = defineProps({
+  initialTitle: String,
+  initialDescription: String,
+  canEdit: Boolean,
+  isDisabled: Boolean
+})
+
+const emit = defineEmits(['update:title', 'update:description'])
```

Comment:
We don't have to use defineModel, but the prop & event names should at least be connected. Rename initialTitle and initialDescription to title & description respectively.